lib/booking_system.rb

```diff
@@ -0,0 +1,156 @@
+require_relative 'reservation.rb'
+require_relative 'block_reservation'
+module Hotel
+  class BookingSystem
+    attr_reader :rooms, :reservations
+    def initialize(room_file = 'data/rooms.csv',
+                   reservation_file = 'data/reservations.csv',
+                   blockreservation_file = 'data/block_reservations.csv')
+      @rooms = load_rooms(room_file)
+      @reservations = load_reservations(reservation_file)
+      @block_reservations = load_block_reservations(blockreservation_file)
+    end
+
+
+    def load_rooms(filename)
+      all_rooms = []
+      CSV.read(filename, headers: true).each do |line|
+        input = {}
+        input[:id] = line[0]
+        input[:cost] = line[1]
+
+        all_rooms << Room.new(input)
+      end
+
+      return all_rooms
+    end
+
+    def load_reservations(filename)
+      all_reservations = []
+      CSV.read(filename, headers: true)
+                         .each do |line|
+        input = {}
+        input[:id] = line[0]
+        input[:guest_name] = line[1]
+        input[:included_rooms] = line[2].split(';').map { |num| num.to_i }
+        input[:rsv_start] = line[3]
+        input[:rsv_end] = line[4]
+
+        all_reservations << Reservation.new(input)
+      end
+
+      return all_reservations
+    end
+
+    def load_block_reservations(filename)
+      all_blocks = []
+      CSV.read(filename, headers: true).each do |line|
+        input = {}
+        input[:id] = line[0]
+        input[:guest_name] = line[1]
+        input[:included_rooms] = line[2].split(';').map { |num| num.to_i }
+        input[:rsv_start] = line[3]
+        input[:rsv_end] = line[4]
+        input[:group_name] = line[5]
+        input[:room_qty] = line[6]
+
+        all_blocks << BlockReservation.new(input)
+      end
+
+      return all_blocks
+    end
+
+    # method to find a reservation based on an inspect date 'insp_date'
+    def find_reservations_by_date(inspect_date)
+      # standardize input date
+      search_date = Date.parse(inspect_date)
+      # select all reservation instances that have booked dates that match
+      # the inspect date
+
+      found_reservations = @reservations.select do |reservation|
+        reservation.booked_dates.include? search_date.to_s
+      end
+      # search through block id's if not found in standard reservations
+      if found_reservations.empty?
```

Comment:
Why do this if it doesn't find reservations during that date.  Instead you would **always** want to find both.